packages/console/src/pages/Connectors/components/ConnectorForm/utils.ts

```diff
@@ -0,0 +1,24 @@
+import type { ConnectorConfigFormItem } from '@logto/connector-kit';
+import { ConnectorConfigFormItemType } from '@logto/connector-kit';
+
+import type { ConnectorFormType } from '../../types';
+
+export const parseFormConfig = (data: ConnectorFormType, formItems: ConnectorConfigFormItem[]) => {
+  return Object.fromEntries(
+    Object.entries(data)
+      .map(([key, value]) => {
+        const formItem = formItems.find((item) => item.key === key);
+
+        if (!formItem) {
+          return null;
+        }
+
+        if (formItem.type === ConnectorConfigFormItemType.Json) {
+          return [key, JSON.stringify(value)];
+        }
+
+        return [key, value];
+      })
+      .filter((item): item is [string, string | number | boolean] => Array.isArray(item))
```

Comment:
Simply filter( (item) => Bool(item)) will work?

Reply:
Then our linter will auto remove `item is [string, unknown]`